src/main/java/org/opensearch/knn/index/query/parser/MethodParametersParser.java

```diff
@@ -0,0 +1,100 @@
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.knn.index.query.model.AlgoQueryParameters;
+import org.opensearch.knn.index.query.model.HNSWAlgoQueryParameters;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER;
+import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER_EF_SEARCH;
+import static org.opensearch.knn.index.IndexUtil.isClusterOnOrAfterMinRequiredVersion;
+import static org.opensearch.knn.index.query.KNNQueryBuilder.EF_SEARCH_FIELD;
+import static org.opensearch.knn.index.query.KNNQueryBuilder.NAME;
+import static org.opensearch.knn.index.query.parser.KNNXParserUtil.parseJsonObject;
+
+@Getter
+public class MethodParametersParser {
+
+    private static final Set<String> VALID_METHOD_PARAMETERS = ImmutableSet.of(METHOD_PARAMETER_EF_SEARCH);
+
+    @SneakyThrows(IOException.class)
+    public static AlgoQueryParameters streamInMethodParameters(StreamInput in) {
+        Integer efSearch = null;
+        if (isClusterOnOrAfterMinRequiredVersion(METHOD_PARAMETER_EF_SEARCH)) {
+            efSearch = in.readOptionalInt();
+        }
+        return efSearch != null ? HNSWAlgoQueryParameters.builder().efSearch(efSearch).build() : null;
+    }
+
+    @SneakyThrows(IOException.class)
+    public static void streamOutMethodParameters(final StreamOutput out, final AlgoQueryParameters algoQueryParameters) {
+        final Optional<HNSWAlgoQueryParameters> hnswMethodParameters = extractHnswAlgoParameters(algoQueryParameters);
+
+        if (isClusterOnOrAfterMinRequiredVersion(METHOD_PARAMETER_EF_SEARCH)) {
+            // Write false even if it doesn't cast to hnsw, so we can deserialize w/o ambiguity as parameters get added
+            out.writeOptionalInt(hnswMethodParameters.flatMap(HNSWAlgoQueryParameters::getEfSearch).orElse(null));
+        }
+    }
+
+    public static Optional<HNSWAlgoQueryParameters> extractHnswAlgoParameters(final AlgoQueryParameters algoQueryParameters) {
+        return Optional.ofNullable(algoQueryParameters)
+            .filter(HNSWAlgoQueryParameters.class::isInstance)
+            .map(HNSWAlgoQueryParameters.class::cast);
+    }
+
+    @SneakyThrows(IOException.class)
+    public static AlgoQueryParameters fromXContent(final XContentParser parser) {
+        final Map<String, Object> parameters = parseJsonObject(parser);
+        if (parameters.isEmpty()) {
+            throw new IllegalArgumentException("[" + NAME + "] method_parameter cannot be empty");
+        }
+
+        for (String jsonkey : parameters.keySet()) {
+            if (!VALID_METHOD_PARAMETERS.contains(jsonkey)) {
+                throw new IllegalArgumentException("[" + NAME + "] unknown parameter " + jsonkey + " found.");
+            }
+        }
+
+        return Optional.ofNullable((Integer) parameters.get(METHOD_PARAMETER_EF_SEARCH))
```

Comment:
This good for 1 single attribute. But we might have to add more validations once we have more attributes that are getting parsed. I would recommend adding a todo here to handle that condition.

Reply:
Yeah the intention is to build on this when nProbs is added